lib/resource-mapper.js

```diff
@@ -91,22 +91,28 @@ class ResourceMapper {
     if (filePath.indexOf('/..') >= 0) {
       throw new Error('Disallowed /.. segment in URL')
     }
-    let isIndex = searchIndex && filePath.endsWith('/')
+    let isFolder = filePath.endsWith('/')
+    let isIndex = searchIndex && isFolder
 
     // Create the path for a new file
     let path
     if (createIfNotExists) {
       path = filePath
-      // Append index filename if needed
-      if (isIndex) {
-        if (contentType !== this._indexContentType) {
-          throw new Error(`Index file needs to have ${this._indexContentType} as content type`)
-        }
-        path += this._indexFilename
-      }
-      // If the extension is not correct for the content type, append the correct extension
-      if (searchIndex && this._getContentTypeByExtension(path) !== contentType) {
-        path += `$${contentType in extensions ? `.${extensions[contentType][0]}` : '.unknown'}`
+      // create a new folder
+      if (!searchIndex && isFolder) {
+  	  // else create a new file
+      } else {
```

Comment:
`if` statement with empty body is to be avoided

Reply:
I shall replace with

> if ((!searchIndex && isFolder) === false) {

What shall I do with the tests ? Are the errors related to the fix ?

Reply:
Or just `if (searchIndex !! !isFolder)`.

Tests are indeed related; some behavior is changed, so should be updated. I think the behavior is correct though.

Reply:
I suppose you meant `if (searchIndex || !isFolder)` which do not cover `searchIndex: false` and `isFolder: false` which would allow to create a file without contentType control or read without check it exists.

I think there are missing tests that should cover the case : 

     itMapsUrl(mapper, "a URL with a bogus extension that doesn't match the content type",
      {
        url: 'http://localhost/space/foo.ttl',
        contentType: 'text/html',
        createIfNotExists: true,
        searchIndex: false
      },
      {
        path: `${rootPath}space/foo.ttl$.html`,
        contentType: 'text/html'
      })

and cases without `createIfNotExists`.

Reply:
> I suppose you meant `if (searchIndex || !isFolder)`

No, I meant that `if ((!searchIndex && isFolder) === false) {` is written more simply as `if (searchIndex || !isFolder)`.

Case looks good.